src/type_object.rs

```diff
@@ -96,6 +96,12 @@ pub unsafe trait PyTypeInfo: Sized + HasPyGilRef {
         }
     }
 
+    /// Returns the safe abstraction over the type object or an error if initialization fails
+    #[inline]
+    fn try_type_object_bound(py: Python<'_>) -> PyResult<Bound<'_, PyType>> {
+        Ok(Self::type_object_bound(py))
+    }
```

Comment:
While I think there's potentially value in having a fallible way to get the type object on this trait, I think that's probably a design question worth having in its own PR. 

Since the previous week I think `LazyTypeObject` now uses the Bound API, so I think you should be able to use that again without needing this here.

Reply:
> While I think there's potentially value in having a fallible way to get the type object on this trait, I think that's probably a design question worth having in its own PR.

Agreed

> Since the previous week I think LazyTypeObject now uses the Bound API, so I think you should be able to use that again without needing this here.

I have rebased my MR and I am using it in dc81d768e61abe180bd84322769bfbb60c6d98a8

About why this new function, my reasoning is: In the `add_to_module` function implemented on types I need to find a way to call `LazyTypeObject::get_or_try_init` on `pyclass` derivation to properly propagate errors while calling the usual `PyTypeInfo::type_object_bound` on other types.
I see two ways to archive this:
- make `#[pyclass]` generate a static `add_to_module` function like `#[pymodule]` and `#[pyfunction]` and implement the `PyAddToModule` trait on all types that are not created via `#[pyclass]` (custom exceptions...). To archive this easily I need a trait that is implemented by everyone but types created by `#[pyclass]`. I have not found such trait and was a bit reluctant to implement it.
- The approach in dc81d768e61abe180bd84322769bfbb60c6d98a8 you have reviewed and that I have reverted.

If the new MR state without dc81d768e61abe180bd84322769bfbb60c6d98a8 is fine with you we can maybe merge it and figure out a way to call `LazyTypeObject::get_or_try_init` in a follow up.